Remove support for CUDA 8

syed-ahmed opened the pull request:
1.1.0 stopped support for CUDA 8

syed-ahmed committed:
Remove instances of cuda 8

syed-ahmed committed:
Restore space

ezyang commented on aten/src/ATen/native/cuda/SortingRadixSelect.cuh:
```diff
@@ -121,7 +121,7 @@ struct TopKTypeConfig<at::Half> {
 
   static inline __device__ RadixType convert(at::Half v) {
-#if CUDA_VERSION >= 8000 || defined __HIP_PLATFORM_HCC__
+#if CUDA_VERSION >= 9000 || defined __HIP_PLATFORM_HCC__
```
Shouldn't you just delete this half of the condition now?

syed-ahmed replied:
If I delete the half, the following won't be enabled when compiling with __CUDACC__?
```
RadixType x = __half_as_ushort(v);
RadixType mask = -((x >> 15)) | 0x8000;
return (v == v) ? (x ^ mask) : 0xffff;
```

ezyang replied:
Oh, I guess it should just be a test if you're compiling with `__CUDACC__`. I guess this is fine (the nit I have is that it implies that this code only works with CUDA 9.0, and that's not really true.)

syed-ahmed replied:
changed to `__CUDA_ARCH__` since that is guaranteed to be defined in device code.

ezyang commented:
Good show, but the `.circleci/config.yml` changes are done incorrectly, as evidenced by https://travis-ci.org/pytorch/pytorch/jobs/530092397 . You need to apply your changes to the original source files inside .circleci, e.g., verbatim-configs. Bug @kostmo if you can't figure out how to do it (and make him write more docs for it :)

ezyang reviewed (changes requested):
failing travis

syed-ahmed commented:
> Good show, but the `.circleci/config.yml` changes are done incorrectly, as evidenced by https://travis-ci.org/pytorch/pytorch/jobs/530092397 . You need to apply your changes to the original source files inside .circleci, e.g., verbatim-configs. Bug @kostmo if you can't figure out how to do it (and make him write more docs for it :)

Totally didn't read the README.md XD. Good that the ci script is now generated. I'll modify accordingly.

syed-ahmed committed:
Use script generator for circleci

ezyang commented:
@pytorchbot rebase this please

Commit:
Merge remote-tracking branch 'origin/master' into HEAD

ezyang commented:
The NUMA build failure looks concerning but I don't see how it could be related to this PR.

```
May 10 13:55:46 [ 97%] Building CXX object CMakeFiles/c10.dir/util/numa.cpp.o
May 10 13:55:47 /var/lib/jenkins/workspace/c10/util/numa.cpp:6:18: fatal error: numa.h: No such file or directory
```

ezyang commented:
@pytorchbot rebase this please

Commit:
Merge remote-tracking branch 'origin/master' into HEAD

syed-ahmed committed:
Merge remote-tracking branch 'upstream/master' into remove-cuda8

syed-ahmed committed:
Use __CUDA_ARCH__ instead of version number for device code

ezyang commented:
The NUMA problem appears to be durable, and related to this PR.

ezyang commented:
My guess it's related to when you changed the docker image for a few jobs; apparently there is a latent bug in a docker configuration we aren't currently testing. You might be able to debug by checking out the relevant docker image, check instructions in https://github.com/pytorch/ossci-job-dsl Holler if you need help.

syed-ahmed committed:
Update build.sh to include cuda 8 env setup for cuda 9 containers

syed-ahmed committed:
Satisfy conda cmake comment and include cuda9-py2 build

"Conda cmake will be newer than our supported min version 3.5, so we only do it in two builds that we know should use conda"

syed-ahmed committed:
Use __CUDA_ARCH__

syed-ahmed commented:
@ezyang I think the problem was some apt-get got disabled because of my removal of some "cuda8" strings. The standalone c10 build test utilizes the cuda9-py3 container now and I backtracked to the cuda8-py3 container CI logs and found that the following apt-get steps were missing. I restored the behavior for cuda 9 in these two commits and think that will fix the CI:
- https://github.com/pytorch/pytorch/pull/20298/commits/61ace2c34e55f4f3a019eba256b3033602adbe3e
- https://github.com/pytorch/pytorch/pull/20298/commits/d7a04623b7202b528f9d8e9632517b237e27caa9
```
ay 10 21:28:27 + sudo apt-get -qq update
May 10 21:28:30 + [[ pytorch-linux-xenial-cuda8-cudnn7-py3-build == *-trusty-py2.7.9* ]]
May 10 21:28:30 + sudo apt-get -qq install --allow-downgrades --allow-change-held-packages openmpi-bin libopenmpi-dev
May 10 21:28:33 debconf: delaying package configuration, since apt-utils is not installed
May 10 21:28:33 Selecting previously unselected package libmagic1:amd64.

May 10 21:28:33 (Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 26586 files and directories currently installed.)

May 10 21:28:33 Preparing to unpack .../libmagic1_1%3a5.25-2ubuntu1.2_amd64.deb ...

May 10 21:28:33 Unpacking libmagic1:amd64 (1:5.25-2ubuntu1.2) ...

May 10 21:28:33 Selecting previously unselected package file.

May 10 21:28:33 Preparing to unpack .../file_1%3a5.25-2ubuntu1.2_amd64.deb ...

May 10 21:28:33 Unpacking file (1:5.25-2ubuntu1.2) ...

May 10 21:28:33 Selecting previously unselected package libltdl7:amd64.

May 10 21:28:33 Preparing to unpack .../libltdl7_2.4.6-0.1_amd64.deb ...

May 10 21:28:33 Unpacking libltdl7:amd64 (2.4.6-0.1) ...

May 10 21:28:33 Selecting previously unselected package libltdl-dev:amd64.

May 10 21:28:33 Preparing to unpack .../libltdl-dev_2.4.6-0.1_amd64.deb ...

May 10 21:28:33 Unpacking libltdl-dev:amd64 (2.4.6-0.1) ...

May 10 21:28:33 Selecting previously unselected package libtool.

May 10 21:28:33 Preparing to unpack .../libtool_2.4.6-0.1_all.deb ...

May 10 21:28:33 Unpacking libtool (2.4.6-0.1) ...

May 10 21:28:33 Selecting previously unselected package libhwloc5:amd64.

May 10 21:28:33 Preparing to unpack .../libhwloc5_1.11.2-3_amd64.deb ...

May 10 21:28:33 Unpacking libhwloc5:amd64 (1.11.2-3) ...

May 10 21:28:33 Selecting previously unselected package libibverbs1.

May 10 21:28:33 Preparing to unpack .../libibverbs1_1.1.8-1.1ubuntu2_amd64.deb ...

May 10 21:28:33 Unpacking libibverbs1 (1.1.8-1.1ubuntu2) ...

May 10 21:28:33 Selecting previously unselected package libopenmpi1.10.

May 10 21:28:33 Preparing to unpack .../libopenmpi1.10_1.10.2-8ubuntu1_amd64.deb ...

May 10 21:28:33 Unpacking libopenmpi1.10 (1.10.2-8ubuntu1) ...

May 10 21:28:33 Selecting previously unselected package openmpi-common.

May 10 21:28:33 Preparing to unpack .../openmpi-common_1.10.2-8ubuntu1_all.deb ...

May 10 21:28:33 Unpacking openmpi-common (1.10.2-8ubuntu1) ...

May 10 21:28:34 Selecting previously unselected package openmpi-bin.

May 10 21:28:34 Preparing to unpack .../openmpi-bin_1.10.2-8ubuntu1_amd64.deb ...

May 10 21:28:34 Unpacking openmpi-bin (1.10.2-8ubuntu1) ...

May 10 21:28:34 Selecting previously unselected package 




:amd64.

May 10 21:28:34 Preparing to unpack .../libnuma-dev_2.0.11-1ubuntu1.1_amd64.deb ...

May 10 21:28:34 Unpacking libnuma-dev:amd64 (2.0.11-1ubuntu1.1) ...

May 10 21:28:34 Selecting previously unselected package libhwloc-dev:amd64.

May 10 21:28:34 Preparing to unpack .../libhwloc-dev_1.11.2-3_amd64.deb ...

May 10 21:28:34 Unpacking libhwloc-dev:amd64 (1.11.2-3) ...

May 10 21:28:34 Selecting previously unselected package ocl-icd-libopencl1:amd64.

May 10 21:28:34 Preparing to unpack .../ocl-icd-libopencl1_2.2.8-1_amd64.deb ...

May 10 21:28:34 Unpacking ocl-icd-libopencl1:amd64 (2.2.8-1) ...

May 10 21:28:34 Selecting previously unselected package libhwloc-plugins.

May 10 21:28:34 Preparing to unpack .../libhwloc-plugins_1.11.2-3_amd64.deb ...

May 10 21:28:34 Unpacking libhwloc-plugins (1.11.2-3) ...

May 10 21:28:34 Selecting previously unselected package libibverbs-dev.

May 10 21:28:34 Preparing to unpack .../libibverbs-dev_1.1.8-1.1ubuntu2_amd64.deb ...

May 10 21:28:34 Unpacking libibverbs-dev (1.1.8-1.1ubuntu2) ...

May 10 21:28:34 Selecting previously unselected package libopenmpi-dev.

May 10 21:28:34 Preparing to unpack .../libopenmpi-dev_1.10.2-8ubuntu1_amd64.deb ...

May 10 21:28:34 Unpacking libopenmpi-dev (1.10.2-8ubuntu1) ...

May 10 21:28:34 Processing triggers for libc-bin (2.23-0ubuntu11) ...

May 10 21:28:34 Setting up libmagic1:amd64 (1:5.25-2ubuntu1.2) ...

May 10 21:28:34 Setting up file (1:5.25-2ubuntu1.2) ...

May 10 21:28:34 Setting up libltdl7:amd64 (2.4.6-0.1) ...

May 10 21:28:34 Setting up libltdl-dev:amd64 (2.4.6-0.1) ...

May 10 21:28:34 Setting up libtool (2.4.6-0.1) ...

May 10 21:28:34 Setting up libhwloc5:amd64 (1.11.2-3) ...

May 10 21:28:34 Setting up libibverbs1 (1.1.8-1.1ubuntu2) ...

May 10 21:28:34 Setting up libopenmpi1.10 (1.10.2-8ubuntu1) ...

May 10 21:28:34 Setting up openmpi-common (1.10.2-8ubuntu1) ...

May 10 21:28:34 Setting up openmpi-bin (1.10.2-8ubuntu1) ...

May 10 21:28:34 update-alternatives: using /usr/bin/mpirun.openmpi to provide /usr/bin/mpirun (mpirun) in auto mode

May 10 21:28:34 Setting up libnuma-dev:amd64 (2.0.11-1ubuntu1.1) ...

May 10 21:28:34 Setting up libhwloc-dev:amd64 (1.11.2-3) ...

May 10 21:28:34 Setting up ocl-icd-libopencl1:amd64 (2.2.8-1) ...

May 10 21:28:34 Setting up libhwloc-plugins (1.11.2-3) ...

May 10 21:28:34 Setting up libibverbs-dev (1.1.8-1.1ubuntu2) ...

May 10 21:28:34 Setting up libopenmpi-dev (1.10.2-8ubuntu1) ...

May 10 21:28:34 update-alternatives: using /usr/lib/openmpi/include to provide /usr/include/mpi (mpi) in auto mode

May 10 21:28:34 Processing triggers for libc-bin (2.23-0ubuntu11) ...

May 10 21:28:34 + sudo apt-get -qq install --no-install-recommends openssh-client openssh-server
May 10 21:28:37 debconf: delaying package configuration, since apt-utils is not installed
May 10 21:28:37 Selecting previously unselected package libwrap0:amd64.

```
